Sortable dependencies

tmcw opened the pull request:
This is admittedly a little far from module-deps's original purpose, but just in case it seems like an acceptable PR:

This change adds a `sortKey` property on each dependency that permits the dependencies to be sorted in 'the order written' - a pre-order traversal of the dependency tree. My usecase is documentation.js, where I want to crawl dependencies but allow the developer control over the output order of documentation.

substack commented:
This seems ok if we can also get some accompanying tests and a note in the documentation. Otherwise this feature seems like it could get broken accidentally in a regression.

tmcw commented:
Great: updated the tests to pass and added a section to the readme.

This also includes a fix for node v6: the behavior of `path.dirname` was changed so that in:

node v5:

```js
require('path').dirname(undefined); // -> undefined
```

node v6:

```js
require('path').dirname(undefined); // -> throws an error
```

This keeps that from throwing an error.

tmcw commented:
Sorry, I won't be able to finish this PR.
